daffodil-io/src/main/scala/org/apache/daffodil/io/DirectOrBufferedDataOutputStream.scala

```diff
@@ -822,7 +843,7 @@ class DirectOrBufferedDataOutputStream private[io] (
           bufOS.flush
           val nBitsPut = directDOS.putFile(
             bufOS.tempFile.toPath,
-            bufOS.nBytes * 8, chunkSizeInBytes, finfo)
+            bufferNBits.toLong, chunkSizeInBytes, finfo)
           nBitsPut / 8
         } else
           directDOS.putBytes(bufOS.getBuf, 0, nBytes, finfo)
```

Comment:
Comment on the context below. We call directDOS.putFile, writing the whole contents of the file. But that doesn't take into account the length of the blob, is that correct? I'm wondering if this logic was broken before and just not tested?

Reply:
My last commit had this mistake, but the original Blob code did not.  I'm making this a little more explicit by having an existingFileLengthInBits value that gets passed in, which would be used for the blob length.

Reply:
I'm wondering if we need that at all. The DOS keeps tracking of the relative bit position of this DOS via ``relBitPos0b``. That is effectively the number of bits that have been written to this DOS. And this value is also set whent he blob DOS is created. Seems like that value can always be used to determine how many bits should be delivered from the underlying output stream, wheter it be file based or byte array based. Maybe that simplifies things a bit?

Reply:
I have made the appropriate changes to just use the relBitPos0b instead of trying to keep track of the existing file length via function parameters.